### PR TITLE
Update docker script

### PR DESCRIPTION
**Pull request by keithc-ca**
* drop support for Ubuntu 16.04
* use apt to install gcc/g++
* update set of supported JDK versions
* update CUDA version (9.0 is no longer available; docker image tag and directory name are different)

Fixes: https://github.com/eclipse/omr/issues/6732.
Fixes: https://github.com/eclipse-openj9/openj9/issues/17039.

### CODE REVIEW

File: buildenv/docker/mkdocker.sh [hunk 1]
```diff
@@ -526,9 +524,12 @@ fi
...
   echo "    curl \\"
   echo "    file \\"
-if [ $arch != s390x ] ; then
-  echo "    g++-7 \\"
-  echo "    gcc-7 \\"
+if [ $version = 18.04 ] ; then
+  echo "    g++-8 \\"
+  echo "    gcc-8 \\"
+else
+  echo "    g++-10 \\"
```
Comment (pshipton) [inline buildenv/docker/mkdocker.sh:531]: Do you want to use 11? I've used it and it has worked to compile all versions.

Comment (pshipton) [reply buildenv/docker/mkdocker.sh:531]: Although I'm not sure if it's available on ub20, which could be why you didn't use it.

File: buildenv/docker/mkdocker.sh [hunk 2]
```diff
@@ -582,11 +580,7 @@ if [ $version = 20.04 ] ; then
   echo "    libnftables1 \\"
 fi
   echo "    libprotobuf-c1 \\"
```
Comment (pshipton) [inline buildenv/docker/mkdocker.sh:582]: Incidentally, we don't directly need protobuf any more, although we did in the past. Can we remove it (maybe in another PR)? There are a number of references to protobuf in this file.

File: buildenv/docker/mkdocker.sh (hunk 1 above)
Comment (keithc-ca) [reply buildenv/docker/mkdocker.sh:531]: It wasn't available on Ubuntu 20.04 when I checked.

Comment (keithc-ca) [reply buildenv/docker/mkdocker.sh:531]: It is available with Ubuntu:22.04, but the script needs other changes for that.

File: buildenv/docker/mkdocker.sh (hunk 2 above)
Comment (keithc-ca) [reply buildenv/docker/mkdocker.sh:582]: Could you create an issue for that, listing protobuf and anything else you think is now unnecessary?

**Review by pshipton — APPROVED** [review]

**Pull request merged**